haskell/test/Lorentz/Contracts/Test/FA2.hs

```diff
@@ -904,4 +911,31 @@ fa2Spec fa2Originate = do
               (#amount 10)
 
           withSender commonOperator $ lCallEP fa2contract (Call @"Transfer") transfers
-          validate $ Left (lExpectAnyMichelsonFailed fa2contract)
+          validate $ Left fa2ReceiverHookUndefined
+
+fa2TokenUndefined :: ExecutorError -> Bool
+fa2TokenUndefined = lExpectFailWith (== [mt|TOKEN_UNDEFINED|])
+
+fa2InsufficientBalance :: ExecutorError -> Bool
+fa2InsufficientBalance = lExpectFailWith (== [mt|INSUFFICIENT_BALANCE|])
+
+fa2TxDenied :: ExecutorError -> Bool
+fa2TxDenied = lExpectFailWith (== [mt|TX_DENIED|])
+
+fa2NotOwner :: ExecutorError -> Bool
+fa2NotOwner = lExpectFailWith (== [mt|NOT_OWNER|])
+
+fa2NotOperator :: ExecutorError -> Bool
+fa2NotOperator = lExpectFailWith (== [mt|NOT_OPERATOR|])
+
+fa2ReceiverHookFailed :: ExecutorError -> Bool
+fa2ReceiverHookFailed = lExpectFailWith (== [mt|RECEIVER_HOOK_FAILED|])
+
+fa2SenderHookFailed :: ExecutorError -> Bool
+fa2SenderHookFailed = lExpectFailWith (== [mt|SENDER_HOOK_FAILED|])
+
+fa2ReceiverHookUndefined :: ExecutorError -> Bool
+fa2ReceiverHookUndefined = lExpectFailWith (== [mt|RECEIVER_HOOK_UNDEFINED|])
+
+fa2SenderHookUndefined :: ExecutorError -> Bool
+fa2SenderHookUndefined = lExpectFailWith (== [mt|SENDER_HOOK_UNDEFINED|])
```

Comment:
This is fine for now, but when we move these tests to `morley-ledgers-test` (and treat them as generic FA2 tests) we should update this check because according to FA2 an error can be a string or a pair where the first item is a string.